app/models/report.rb

```diff
@@ -99,6 +102,17 @@ def rules
     Rule.with_discarded.where(id: rule_ids)
   end
 
+  def forwardable?
+    !target_account.local?
+  end
+
+  def forwardable_to_domains
+    target_domain = forwardable? ? [target_account.domain] : []
+
+    status_domains = Account.remote.where(id: Status.where(id: status_ids).where.not(in_reply_to_account_id: nil).select(:in_reply_to_account_id)).pluck(:domain)
+    target_domain.concat(status_domains).uniq
+  end
```

Comment:
The queries that this makes are kind of inefficient, I'm not sure how they could be improved? 

@mjankowski any ideas from a more experienced rails developer? This is based on the logic in https://github.com/mastodon/mastodon/blob/main/app/services/report_service.rb#L62

Reply:
I suspect this should probably also be all the mentioned accounts' domains maybe?

Reply:
Note: I made a few changes, but overall the result is the same.